### PR TITLE
fixed the example spec with neo4j scheme

### PR DESCRIPTION
**Pull request by Kinju011**

### CODE REVIEW

File: lib/neo4j/driver/exceptions/socket_exception.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+module Neo4j
+  module Driver
+    module Exceptions
+      # While resolving an address using getaddrinfo method of Addrinfo it raises an error of SocketError
+      class SocketException < Neo4jException
+      end
+    end
+  end
+end
```
Comment (klobuczek) [inline lib/neo4j/driver/exceptions/socket_exception.rb]: let's don't introduce new Neo4j exception but just catch SocketError below

File: ruby/neo4j/driver/internal/cluster/rediscovery_impl.rb [hunk 2]
```diff
@@ -49,6 +49,7 @@ def resolve
             resolve_all_by_domain_name(server_address).unicast_stream
             # rescue java.net.UnknownHostException => e
           rescue => e
+            e = e.is_a?(SocketError) ? Exceptions::SocketException.new : e
             exception&.add_suppressed(e)
```
Comment (klobuczek) [inline ruby/neo4j/driver/internal/cluster/rediscovery_impl.rb]:
```suggestion
          rescue SocketError => e
```
let's just store the 1st exception. There is not much info in those exceptions anyway.

File: spec/neo4j/driver/dev_manual_examples_spec.rb [hunk 3]
```diff
@@ -34,8 +34,8 @@
 
     context 'Example 2.3. Custom Address Resolver' do
```
Comment (klobuczek) [inline spec/neo4j/driver/dev_manual_examples_spec.rb]: let's add something like
```
      let(:uri) { 'neo4j://nonexisting' }
```
so the spec is effective regardless of the global scheme.

**Review by klobuczek — CHANGES_REQUESTED** [review]

File: lib/neo4j/driver/exceptions/socket_exception.rb (hunk 1 above)
Comment (Kinju011) [reply lib/neo4j/driver/exceptions/socket_exception.rb]: okay

**Comment by klobuczek** [comment]
@Kinju011 the spec in this PR was not testing properly, that's why the spec was passing. The spec should be passing in the current form after my modification. Unfortunately this is not happening even for jruby.

**Comment by klobuczek** [comment]
had to move the spec cluster spec as the resolver is effective only in cluster

**Review by klobuczek — APPROVED** [review]

**Pull request merged**